src/main/java/com/patsi/controller/SocialMediaUserController.java

```diff
@@ -37,7 +41,10 @@ public SocialMediaUser createSocialMediaAccount(@RequestBody SocialMediaUser use
     }
 
     @PostMapping("/getUserByToken")
-    public SocialMediaUser getUserById(@RequestHeader("Authorization") String token) {
+    @RequireLoginSession
+    public SocialMediaUser getUserById() {
+        String token = (String) request.getAttribute("token");
```

Comment:
Extract `request.getAttribute("token");` to a service. You may add it to the existing UserProfileService, or create a new service such as UserSessionService.